src/Aspire.Dashboard/Components/Pages/IPageWithSessionAndUrlState.cs

```diff
@@ -56,13 +57,24 @@ public static class PageExtensions
     /// Called after a change in the view model that will affect the url associated with new page state
     /// to navigate to the new url and save new state in localstorage.
     /// </summary>
-    public static async Task AfterViewModelChangedAsync<TViewModel, TSerializableViewModel>(this IPageWithSessionAndUrlState<TViewModel, TSerializableViewModel> page) where TSerializableViewModel : class
+    public static async Task AfterViewModelChangedAsync<TViewModel, TSerializableViewModel>(this IPageWithSessionAndUrlState<TViewModel, TSerializableViewModel> page, AspirePageContentLayout? layout, bool isChangeInToolbar) where TSerializableViewModel : class
     {
-        var serializableViewModel = page.ConvertViewModelToSerializable();
-        var pathWithParameters = page.GetUrlFromSerializableViewModel(serializableViewModel).ToString();
+        if (layout is not null && !layout.ViewportInformation.IsDesktop && isChangeInToolbar)
```

Comment:
Add a comment about what this does

Reply:
Isn't AfterViewModelChangedAsync potentially called multiple times on a page? If so, won't this func be added to DialogCloseListeners multiple times?

Reply:
Yep, thank you for catching this. I have made it so that only the last listener here will be added